### PR TITLE
languages: added elvish

### PR DESCRIPTION
**Pull request by rsteube**
Added [elvish](https://elv.sh/).  Can't see the grammar taking any effect so there is likely still sth. wrong with the config.

### CODE REVIEW

**Comment by the-mikedavis** [comment]
For the syntax highlighting you'll want to add `runtime/queries/elvish/highlights.scm` and `runtime/queries/elvish/injections.scm` following the queries in tree-sitter-elvish: https://github.com/ckafi/tree-sitter-elvish/tree/main/queries

The captures need to be adapted to the ones helix uses though: https://docs.helix-editor.com/master/themes.html#syntax-highlighting, for example `@parameter` should become `@variable.parameter`

And the `injections.scm` in tree-sitter-elvish looks like it's in neovim format. In helix format that injection would be

```scm
((comment) @injection.content
 (#set! injection.language "comment"))
```

**Comment by the-mikedavis** [comment]
Also to clear up the Docs CI's ❌, you'll want to run `cargo xtask docgen` and commit the changes

**Comment by rsteube** [comment]
> For the syntax highlighting you'll want to add `runtime/queries/elvish/highlights.scm` and `runtime/queries/elvish/injections.scm` following the queries in tree-sitter-elvish: https://github.com/ckafi/tree-sitter-elvish/tree/main/queries
> 
> The captures need to be adapted to the ones helix uses though: https://docs.helix-editor.com/master/themes.html#syntax-highlighting, for example `@parameter` should become `@variable.parameter`
> 
> And the `injections.scm` in tree-sitter-elvish looks like it's in neovim format. In helix format that injection would be
> 
> ```scheme
> ((comment) @injection.content
>  (#set! injection.language "comment"))
> ```

So basically that probably needs to be forked to support neovim and helix (different format)?

**Comment by the-mikedavis** [comment]
We host the queries in this repo (https://github.com/helix-editor/helix/tree/master/runtime/queries) since each editor tends to use their own queries. Technically the injections in that repo are wrong according to `tree-sitter-cli` and could be updated. But for getting syntax highlighting working for elvish in helix we just need the queries to be adapted in-repo here

**Comment by rsteube** [comment]
Not sure if i got them all, but syntax highlighting is working now and i don't see an error.

File: runtime/queries/elvish/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+(while (else "else" @keyword.control.repeat))
+(for "for" @keyword.control.repeat)
+(for (else "else" @keyword.control.repeat))
+
+(try "try" @keyword.control.exception)
+(try (catch "catch" @keyword.control.exception))
+(try (else "else" @keyword.control.exception))
+(try (finally "finally" @keyword.control.exception))
+
+(import "use" @include)
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:20]:
```suggestion
(import "use" @keyword.control.import)
```

File: runtime/queries/elvish/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+(import (bareword) @string.special)
+
+(wildcard ["*" "**" "?"] @string.special)
+
+(command argument: (bareword) @variable.parameter)
+(command head: (identifier) @function)
+((command head: (identifier) @keyword.return)
+ (#eq? @keyword.return "return"))
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:28]:
```suggestion
((command head: (identifier) @keyword.return)
 (#eq? @keyword.return "return"))
```
```suggestion
((command head: (identifier) @keyword.control.return)
 (#eq? @keyword.control.return "return"))
```

File: runtime/queries/elvish/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+((command head: _ @function)
+ (#any-of? @function
+  "+" "-" "*" "/" "%" "<" "<=""==" "!=" ">"
+  ">=" "<s" "<=s" "==s" "!=s" ">s" ">=s"
+))
+
+(pipeline "|" @operator)
+(redirection [">" "<" ">>" "<>"] @operator)
+
+(io_port) @number
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:40]:
```suggestion
(io_port) @constant.numeric
```

File: runtime/queries/elvish/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+(temporary_assignment
+  "tmp" @keyword
+  (lhs (identifier) @variable))
+
+(variable_deletion
+  "del" @keyword
+  (identifier) @variable)
+
+
+(number) @number
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:66]:
```suggestion
(number) @constant.numeric
```

File: runtime/queries/elvish/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+(string) @string
+
+(variable (identifier) @variable)
+((variable (identifier) @function)
+  (#match? @function ".+\\~$"))
+((variable (identifier) @boolean)
+ (#any-of? @boolean "true" "false"))
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:73]:
```suggestion
((variable (identifier) @constant.builtin.boolean)
 (#match? @constant.builtin.boolean "(true|false)"))
```

We don't have `#any-of?` implemented at the moment so we use equivalent `#match?`s

File: runtime/queries/elvish/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+((variable (identifier) @constant.builtin)
+ (#any-of? @constant.builtin
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:75]: same here about `#any-of?`

File: runtime/queries/elvish/highlights.scm [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+(command head: (identifier) @function)
+((command head: (identifier) @keyword.return)
+ (#eq? @keyword.return "return"))
+((command (identifier) @keyword.operator)
+ (#any-of? @keyword.operator "and" "or" "coalesce"))
+((command head: _ @function)
+ (#any-of? @function
+  "+" "-" "*" "/" "%" "<" "<=""==" "!=" ">"
+  ">=" "<s" "<=s" "==s" "!=s" ">s" ">=s"
+))
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:35]: and here with `#any-of?`

File: runtime/queries/elvish/highlights.scm [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+(variable_deletion
+  "del" @keyword
+  (identifier) @variable)
+
+
+(number) @number
+(string) @string
+
+(variable (identifier) @variable)
```
Comment (the-mikedavis) [inline runtime/queries/elvish/highlights.scm:69]: There's a snag here where this stanza is higher precedence than the ones below it so the ones below will never match. If you move this down below the other `(variable (identifier) ...)` stanzas you'll see highlights for booleans for example

Comment (the-mikedavis) [reply runtime/queries/elvish/highlights.scm:62]: Oh actually I meant the `(variable (identifier) @variable)` stanza. The GitHub UI makes it look like I'm talking about the number/string stanzas 😅

**Review by frexsdev — APPROVED** [review]

**Review by the-mikedavis — COMMENTED** [review]
Could you re-run `cargo xtask docgen`? It should add a checkmark to the highlight column for elvish. After that I think this should be good to go 👍

**Comment by rsteube** [comment]
Sure. Will do when i'm back home.

**Review by the-mikedavis — APPROVED** [review]
Thanks for working on this, this looks great!

**Pull request merged**